### PR TITLE
Seeed XIAO support

### PR DESCRIPTION
**Pull request by sago35**
I added Seeed XIAO (Seeeduino XIAO) support to TinyGo.

The SEEED Seeeduino XIAO is a minimal, low-cost board that uses the Atmel ATSAMD21G18,
a 32-bit ARM Cortex-M0+ processor with 256KB flash and 32KB SRAM running at 48MHz.
The board is 20 x 17.5mm, and uses a USB-C connector.

The price is only $4.9.
It's more expensive than bluepill but considerably cheaper than trinket-m0 ($8.95).
It doesn't have external flash, but it's SAMD21, so it's a good first step for TinyGo.
https://wiki.seeedstudio.com/Seeeduino-XIAO/

ATSAMD2x datasheet: https://files.seeedstudio.com/wiki/Seeeduino-XIAO/res/ATSAMD21G18A-MU-Datasheet.pdf
PinOut: https://files.seeedstudio.com/wiki/Seeeduino-XIAO/img/Seeeduino-XIAO-pinout.jpg
Schematics: https://files.seeedstudio.com/wiki/Seeeduino-XIAO/res/Seeeduino-XIAO-v1.0-SCH-191112.pdf

* Feature
    * ATSAMD21G18 (48MHz / 48 Pins / 256KB / 32KB)
    * SWDIO / SWCLK
    * 2 blue LED + 1 yello LED
    * Type-C interface

* TODO
    * [x] RESET_MAGIC_VALUE
    * [x] USB CDC identifiers / VID / PID
    * [x] Check: ./src/examples
    * [x] Check: tinygo.org/x/drivers/examples
    * [x] Update README.md
    * [x] Update Makefile

### CODE REVIEW

**Comment by sago35** [comment]
board info from Arduino IDE
image

**Comment by sago35** [comment]
image

File: src/machine/board_xiao.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	A8  = PA07 // ADC/AIN[7]
+	A9  = PA05 // ADC/AIN[6]
+	A10 = PA06 // ADC/AIN[5]
+)
+
+const (
+	LED         = PA17
+	PIN_LED_13  = PA17
+	PIN_LED     = PA17
+	LED_BUILTIN = PA17
```
Comment (aykevl) [inline src/machine/board_xiao.go]: `LED_BUILTIN` is used in the Arduino environment. I think we should only define `LED` here (not the other three definitions), as that is the convention for other boards within TinyGo.

File: src/machine/board_xiao.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+	PIN_LED_RXL = PA18
+	PIN_LED_TXL = PA19
+	PIN_LED2    = PIN_LED_RXL
+	PIN_LED3    = PIN_LED_TXL
```
Comment (aykevl) [inline src/machine/board_xiao.go]: Can you match the TinyGo convention here?

```suggestion
	LED_RXL = PA18
	LED_TXL = PA19
	LED1    = LED
	LED2    = LED_RXL
	LED3    = LED_TXL
```

File: src/machine/board_xiao.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+)
+
+const (
+	SWDIO = PA31
+	SWCLK = PA30
+)
```
Comment (aykevl) [inline src/machine/board_xiao.go]: These cannot normally be used from the firmware so there isn't a good reason to define these.

File: targets/xiao.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+    "inherits": ["atsamd21g18a"],
+    "build-tags": ["sam", "atsamd21g18a", "xiao"],
+    "flash-1200-bps-reset": "true",
+    "flash-method": "msd",
+    "msd-volume-name": "ITSYBOOT",
```
Comment (aykevl) [inline targets/xiao.json]: It looks like they just copied the ItsyBitsy-M0 bootloader, which is unfortunate. But we'll have to match that.

File: src/machine/board_xiao.go (hunk 1 above)
Comment (sago35) [reply src/machine/board_xiao.go]: Fixed at 14d94840e950d59847d15abc52a9f57db2dd1361

File: src/machine/board_xiao.go (hunk 2 above)
Comment (sago35) [reply src/machine/board_xiao.go]: Fixed at 14d94840e950d59847d15abc52a9f57db2dd1361

File: src/machine/board_xiao.go (hunk 3 above)
Comment (sago35) [reply src/machine/board_xiao.go]: Removed at af11b388c46a642ecbdde6c5a1992cf6a9ef8473

File: targets/xiao.json (hunk 4 above)
Comment (sago35) [reply targets/xiao.json]: Fixed at 201dfc2aa883091b6bcacd9c31ff6d69e7e9ec74
The volume name was `Arduino` as well as Wio Terminal (#1124).

**Comment by sago35** [comment]
Created #1174 to add some ADC pin settings.

**Comment by sago35** [comment]
Basic testing was done.
The results were fine.

* test ok
    * ./src/examples/blinky1
    * ./src/examples/adc
    * ./src/examples/button
    * ./src/examples/echo
    * ./src/examples/serial
    * ./src/examples/pininterrupt
    * ./src/examples/systick
    * ./src/examples/pwm
        * Some of the pins are OK : D10 D8 D5 D4
    * I2C
        * tinygo.org/x/drivers/examples/adt7410
    * SPI
        * tinygo.org/x/drivers/examples/ili9341
    * UART
        * ./src/examples/echo
* not tested
    * I2S
        * I don't have an I2S device.

**Comment by sago35** [comment]
Completed.
Please review.

**Comment by deadprogram** [comment]
Hello @sago35 can you please resolve the merge conflict? Thank you.

**Comment by sago35** [comment]
@deadprogram 
rebased and force-pushed.

I want to merge this PR and then merge #1180.

**Comment by deadprogram** [comment]
Thank you very much for another great contribution @sago35 now squash/merging.

**Pull request merged**